specification/profiling.md

```diff
@@ -0,0 +1,169 @@
+# Profiling Conventions
+
+**Status**: [experimental](../README.md#versioning-and-status-of-the-specification)
+
+This document describes the specific behavior and conventions for Splunk
+agents that contain profiling features.
```

Comment:
Should we not replace `agent` with `instrumentation library` here and in other places

See: https://github.com/signalfx/gdi-specification#terms and the rest of the spec e.g. https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md#instrumentation-libraries

Reply:
Naming is hard. I like this change. 😁

Reply:
My tongue always favors agent because it's 6 syllables shorter!  Heh.